crates/red_knot_python_semantic/src/types.rs

```diff
@@ -854,26 +854,26 @@ impl<'db> Type<'db> {
             }
             (
                 Type::SubclassOf(SubclassOfType {
-                    base: ClassBase::Any,
+                    base: ClassBase::Any | ClassBase::Todo(_) | ClassBase::Unknown,
                 }),
                 Type::SubclassOf(_),
             ) => true,
             (
                 Type::SubclassOf(SubclassOfType {
-                    base: ClassBase::Any,
+                    base: ClassBase::Any | ClassBase::Todo(_) | ClassBase::Unknown,
                 }),
-                Type::Instance(target),
-            ) if target.class.is_known(db, KnownClass::Type) => true,
+                Type::Instance(_),
+            ) if target.is_subtype_of(db, KnownClass::Type.to_instance(db)) => true,
```

Comment:
or should I use `is_assignable_to` here...? I suppose we currently consider all `Instance` types fully static, but that will no longer be the case after we fix https://github.com/astral-sh/ruff/issues/14774?

Reply:
Probably should use `is_assignable`, as it's correct here, and will be more robust in future when `Type::Instance` is not necessarily fully-static.